src/Microsoft.SqlTools.ServiceLayer/LanguageServices/LanguageService.cs

```diff
@@ -443,6 +446,51 @@ await requestContext.SendEvent(TelemetryNotification.Type, new TelemetryParams()
         }
 
         /// <summary>
+        /// Handle the rebuild IntelliSense cache notification
+        /// </summary>
+        public async Task HandleRebuildIntelliSenseNotification(ScriptFile scriptFile, EventContext eventContext)
```

Comment:
Minor: there's a lot of overlap between this and the UpdateLanguageServiceOnConnection method. Could we share some of the logic between them?